### PR TITLE
5004/fix conflict with std

### LINKED ISSUE #5004: [iso4] constrain Jinja
The std module has a constraint on Jinja that is not present on core. Since the compiler environment is activated in-process after this dependency is already loaded in `sys.modules`, core's imcompatible version is used in practice.

- [ ] Constrain Jinja to the same version as it is constrained to in the std module on the iso4 branch
- [ ] Check if there are other dependency conflicts between std and core

### PR DESCRIPTION
**Pull request by FloLey**
# Description

The std module has a constraint on Jinja that is not present on core. Since the compiler environment is activated in-process after this dependency is already loaded in sys.modules, core's imcompatible version is used in practice. This commit will 
constrain Jinja to the same version as it is constrained to in the std module.

closes #5004 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: changelogs/unreleased/5004-constraint-conflict-std.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+description: "constraints Jinja and email_validator to same version as std"
+change-type: patch
+destination-branches:
```
Comment (sanderr) [inline changelogs/unreleased/5004-constraint-conflict-std.yml:3]: Should we add a bugfix entry?

**Review by sanderr — CHANGES_REQUESTED** [review]
We also need to modify the constraint(s?) in `setup.py`.

File: changelogs/unreleased/5004-constraint-conflict-std.yml (hunk 1 above)
Comment (FloLey) [reply changelogs/unreleased/5004-constraint-conflict-std.yml:3]: I would only add a bugfix entry if it ever broke something, but did it?

Comment (sanderr) [reply changelogs/unreleased/5004-constraint-conflict-std.yml:3]: [It did](https://inmanta.slack.com/archives/CKRF0C8R3/p1666012217831119). Only for Python>3.6 though (see [here](https://inmanta.slack.com/archives/CKRF0C8R3/p1666016740737439?thread_ts=1666012217.831119&cid=CKRF0C8R3)).

**Review by sanderr — APPROVED** [review]

**Pull request closed without merging**